src/Fantomas/TokenParser.fs

```diff
@@ -878,6 +884,15 @@ let rec private getTriviaFromTokensThemSelves
 
         getTriviaFromTokensThemSelves mkRange allTokens nextTokens info
 
+    | KeywordString ks :: rest ->
+        let range = getRangeBetween mkRange ks ks
+
+        let info =
+            Trivia.Create(KeywordString(ks.Content)) range
```

Comment:
Capture Keyword strings like `__SOURCE_DIRECTORY__` as a custom trivia type.
They are later assigned as ContentItSelf of a trivia node.